### PR TITLE
[Draft] [Do not review] Push Notifications LRP lifetime V0.1

### PR DESCRIPTION
**Pull request by danielayala94**

### CODE REVIEW

File: dev/PushNotifications/PushNotificationManager.cpp [hunk 1]
```diff
@@ -245,4 +252,43 @@ namespace winrt::Microsoft::Windows::PushNotifications::implementation
             LOG_IF_FAILED(::CoRevokeClassObject(static_cast<DWORD>(token.Cookie())));
         }
     }
+
+    uint32_t PushNotificationManager::GetStringLengthFromComServer(const winrt::hstring& inputString)
+    {
+        try
+        {
+            wil::com_ptr<IWpnLrpPlatform> reunionEndpoint{
```
Comment (hulumane) [inline dev/PushNotifications/PushNotificationManager.cpp:260]: please rename to platform. Its not an endpoint : )

Comment (hulumane) [inline dev/PushNotifications/PushNotificationManager.cpp:260]: wil Vs WRL Vs Winrt : ) Why did we end up choosing wil over the other options?

File: dev/PushNotifications/PushNotificationManager.cpp [hunk 2]
```diff
@@ -245,4 +252,43 @@ namespace winrt::Microsoft::Windows::PushNotifications::implementation
...
+            HRESULT error = wil::ResultFromCaughtException();
+            return error;
+        }
+    }
+
+    uint32_t PushNotificationManager::ShutdownPlatformManually()
+    {
+        try
+        {
+            wil::com_ptr<IWpnLrpPlatform> reunionEndpoint{
```
Comment (hulumane) [inline dev/PushNotifications/PushNotificationManager.cpp:281]: platform

File: dev/PushNotifications/PushNotificationManager.cpp [hunk 3]
```diff
@@ -245,4 +252,43 @@ namespace winrt::Microsoft::Windows::PushNotifications::implementation
...
+            return length;
+        }
+        catch (...)
+        {
+            HRESULT error = wil::ResultFromCaughtException();
+            return error;
+        }
+    }
+
+    uint32_t PushNotificationManager::ShutdownPlatformManually()
```
Comment (hulumane) [inline dev/PushNotifications/PushNotificationManager.cpp:277]: Hmm why is WinSDK doing lifetime management for the LRP?

Comment (danielayala94) [reply dev/PushNotifications/PushNotificationManager.cpp:277]: It is just for testing purposes. The final version won't do this.

**Comment by DrusTheAxe** [comment]
If you're not ready to review you can make a branch but no PR. Other devs can even update the branch too without a PR

Only reason for a PR is if you're ready for review, and only reason for a draft PR is if you want feedback, or you have a complex description you want to write down  even though you're not ready for review

**Pull request closed without merging**